model/task/task.go

```diff
@@ -590,6 +594,36 @@ func (t *Task) MarkGeneratedTasks() error {
 	)
 }
 
+// SetGeneratedJSON sets JSON data to generate tasks from.
+func (t *Task) SetGeneratedJSON(json []json.RawMessage) error {
+	t.GeneratedJSON = json
+	return UpdateOne(
```

Comment:
should we not set this if it's already set?

Reply:
No, we call this function after we've done the work of generating tasks. The only case where that would matter is if there are multiple jobs running simultaneously, which amboy's machinery should (and does) prevent.